corgidrp/spec.py

```diff
@@ -0,0 +1,410 @@
+                                         options={'maxiter': 5000})
+    
+    if not registration_result.success:
+        print(f"Warning: Registration optimization did not converge: {registration_result.message}")
+    
+    # Don't round the template positions
+    xcom_template, ycom_template = xcent_template, ycent_template
+    xcom_data, ycom_data = xcent_guess, ycent_guess
+    
+    xfit = xcom_template + registration_result.x[0]
+    yfit = ycom_template + registration_result.x[1]
+
+    psf_data_bkg = psf_data.copy()
+    psf_data_bkg[ymin_data_cut:ymax_data_cut+1, xmin_data_cut:xmax_data_cut+1] = np.nan
+    psf_peakpix_snr = np.max(psf_data) / np.nanstd(psf_data_bkg)
+    
+    (gauss2d_xfit, gauss2d_yfit, xfwhm, yfwhm, gauss2d_peakfit,
+     fitted_data_stamp, model, residual) = gaussfit2d_pix(psf_data,
+                                                xguess = xfit,
+                                                yguess = yfit,
+                                                xfwhm_guess = fwhm_minor_guess, 
+                                                yfwhm_guess = fwhm_major_guess,
+                                                halfwidth = 1, halfheight = halfheight, 
+                                                guesspeak = np.max(psf_data), oversample = gauss2d_oversample, 
+                                                refinefit = True)
+    
+    (x_precis, y_precis) = (np.abs(xfwhm) / (2 * np.sqrt(2 * np.log(2))) / psf_peakpix_snr,
+                            np.abs(yfwhm) / (2 * np.sqrt(2 * np.log(2))) / psf_peakpix_snr)
+
+    return xfit, yfit, gauss2d_xfit, gauss2d_yfit, psf_peakpix_snr, x_precis, y_precis
+
+def compute_psf_centroid(dataset, initial_cent, output_path=None, verbose=False, halfwidth=10, halfheight=10):
+    """
+    Compute PSF centroids for a grid of PSFs and store them in a calibration file.
+
+    Args:
+        dataset (Dataset): Dataset containing 2D PSF images. Each image must include pri_hdr and ext_hdr.
+        initial_cent (dict): Dictionary with initial guesses for PSF centroids.
+                             Must include keys 'xcent' and 'ycent', each mapping to an array of shape (N,).
+                             Example:
+                                 initial_cent = {
+                                     'xcent': [12.0, 15.5, 10.3],
+                                     'ycent': [8.2, 14.1, 9.9]
+                                 }
+        output_path (str, optional): Path to write the resulting FITS file. Directory must exist.
+        verbose (bool): If True, prints fitted centroid values for each frame.
+        halfwidth (int): Half-width of the PSF fitting box. Default is 10.
+        halfheight (int): Half-height of the PSF fitting box. Default is 10.
+
+    Returns:
+        SpectroscopyCentroidPSF: Calibration object with fitted (x, y) centroids.
+    """
+    from corgidrp.data import Dataset
```

Comment:
very minor, but imports should be done at the start of the module, not inside of functions (unless we have a very good reason for it).